demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,196 @@
+        tracker_action_params.min_track_duration = 8;
+        tracker_action_params.forget_delay = 150;
+        tracker_action_params.affinity_thr = 0.9f;
+        tracker_action_params.averaging_window_size_for_rects = 5;
+        tracker_action_params.averaging_window_size_for_labels = 1;
+        tracker_action_params.bbox_heights_range = cv::Vec2f(10, 2160);
+        tracker_action_params.drop_forgotten_tracks = false;
+        tracker_action_params.max_num_objects_in_track = std::numeric_limits<int>::max();
+        tracker_action = std::make_shared<Tracker>(tracker_action_params);
+    }
+    static void run(const cv::Mat& frame,
+                    const TrackedObjects& actions,
+                          TrackedObjects& tracked_actions,
+                          Tracker& tracker_action) {
+        tracker_action.Process(frame, actions);
+        tracked_actions = tracker_action.TrackedDetectionsWithLabels();
+    }
+};
+
+GAPI_OCV_KERNEL_ST(OCVPreprocessing, custom::ActionRecognisePreprocessing, ARStateParams) {
+    static void setup(const cv::GMatDesc&,
+                      const cv::GArrayDesc&,
+                      const cv::GOpaqueDesc&,
+                      const cv::Scalar&,
+                            std::shared_ptr<ARStateParams>& ar_state) {
+        std::list<cv::Mat> frames = {};
+        ar_state = std::make_shared<ARStateParams>(ARStateParams{frames, 0});
+    }
+
+    static void run(const cv::Mat& in_frame,
+                    const TrackedObjects& tracked_persons,
+                    const std::shared_ptr<int>& runtime_id,
+                    const cv::Scalar& net_size,
+                          std::vector<cv::Mat> &prepared_mat,
+                    ARStateParams& ar_state) {
+        const int channels = int(net_size[0]);
+        const int batch = int(net_size[1]);
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
+        cv::Size sz_dst(height, width);
+        if (ar_state.stored_frames.size() < size_t(batch)) {
+            ar_state.stored_frames.emplace_back(in_frame.clone());
+        } else {
+            if(prepared_mat.size() == 0) {
+                cv::Mat mat_to_process;
+                mat_to_process.create({1, channels, batch, height, width}, CV_32F);
+                // FIXME: Why is GArray result?
+                prepared_mat.push_back(mat_to_process);
+            }
+            ar_state.stored_frames.pop_front();
+            ar_state.stored_frames.push_back(in_frame.clone());
+
+            if (size_t(*runtime_id) < tracked_persons.size()) {
+                ar_state.last_id = *runtime_id;
+            }
+            int step = 0;
+            for(const auto& frame : ar_state.stored_frames) {
+                cv::Mat crop, cvt, resized;
+                if (tracked_persons.size() > 0 &&
+                    size_t(ar_state.last_id) < tracked_persons.size()) {
+                    cv::Rect roi = tracked_persons.at(ar_state.last_id).rect;
```

Comment:
so may this ROI change for an old frame?

Reply:
your track is calculated for the current frame, isn't it? Is it correct to use this ROI on older frames?

Reply:
Yes, it is correct. Person have to stay in the frame.